central/declarativeconfig/transform/transformer.go

```diff
@@ -0,0 +1,34 @@
+package transform
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+// Transformer transforms a declarativeconfig.Configuration to proto.Message(s).
+type Transformer interface {
+	Transform(config declarativeconfig.Configuration) ([]proto.Message, error)
+}
+
+// New creates a Transformer that can handle transforming all currently supported declarativeconfig.Configuration.
+func New() Transformer {
+	return &defaultTransformer{configurationTransformers: map[string]Transformer{
+		declarativeconfig.AuthProviderConfiguration:  nil,
+		declarativeconfig.AccessScopeConfiguration:   nil,
+		declarativeconfig.RoleConfiguration:          nil,
+		declarativeconfig.PermissionSetConfiguration: nil,
+	}}
+}
+
+type defaultTransformer struct {
```

Comment:
nit: IMHO `default` does not tell us what is the meaning of this transformer. Perhaps `combined` or `universal`?